api/app/db/models/forecasts.py

```diff
@@ -42,27 +43,28 @@ class NoonForecast(Base):
     id = Column(Integer, primary_key=True)
     weather_date = Column(TZTimeStamp, nullable=False, index=True)
     station_code = Column(Integer, nullable=False, index=True)
-    temp_valid = Column(Boolean, default=False, nullable=False)
-    temperature = Column(Float, nullable=False)
-    rh_valid = Column(Boolean, default=False, nullable=False)
-    relative_humidity = Column(Float, nullable=False)
-    wdir_valid = Column(Boolean, default=False, nullable=False)
+    temp_valid = Column(Boolean, nullable=False, default=False)
+    temperature = Column(Float, nullable=False, default=math.nan)
+    rh_valid = Column(Boolean, nullable=False, default=False)
+    relative_humidity = Column(Float, nullable=False, default=math.nan)
+    wdir_valid = Column(Boolean, nullable=False, default=False)
     # Set default wind_direction to NaN because some stations don't report it
     wind_direction = Column(Float, nullable=False, default=math.nan)
-    wspeed_valid = Column(Boolean, default=False, nullable=False)
-    wind_speed = Column(Float, nullable=False)
-    precip_valid = Column(Boolean, default=False, nullable=False)
-    precipitation = Column(Float, nullable=False)
+    wspeed_valid = Column(Boolean, nullable=False, default=False)
+    wind_speed = Column(Float, nullable=False, default=math.nan)
+    precip_valid = Column(Boolean, nullable=False, default=False)
+    precipitation = Column(Float, nullable=False, default=math.nan)
     gc = Column(Float, nullable=False, default=math.nan)
     ffmc = Column(Float, nullable=False, default=math.nan)
     dmc = Column(Float, nullable=False, default=math.nan)
     dc = Column(Float, nullable=False, default=math.nan)
     isi = Column(Float, nullable=False, default=math.nan)
     bui = Column(Float, nullable=False, default=math.nan)
     fwi = Column(Float, nullable=False, default=math.nan)
-    danger_rating = Column(Integer, nullable=False)
+    danger_rating = Column(Integer, nullable=True, default=0)
     created_at = Column(TZTimeStamp, nullable=False,
                         default=time_utils.get_utc_now(), index=True)
+    update_date = Column(TZTimeStamp, nullable=True, index=True)
```

Comment:
Yes, the unique constraint enforces this.

Reply:
@conbrad can we please chat about this?

Reply:
Is it really nullable?

Reply:
@Sybrand updated:

Migration first adds this column as nullable, since existing data does not have values here, then sets the column to be equal to the creation_date for existing records, then enforces a non-nullable constraint. All new records will have wfwx_update_date set to the updateDate that WFWX promises to return.